src/main/java/org/traccar/WebDataHandler.java

```diff
@@ -184,12 +186,13 @@ public String formatRequest(Position position) throws UnsupportedEncodingExcepti
         private int retries = 0;
         private Map<String, Object> payload;
         private Invocation.Builder requestBuilder;
+        private MediaType mediaType = MediaType.APPLICATION_JSON_TYPE;
 
         AsyncRequestAndCallback(Position position) {
 
             String formattedUrl;
             try {
-                formattedUrl = json ? url : formatRequest(position);
+                formattedUrl = formatRequest(position);
```

Comment:
There is a performance implication of doing `formatRequest`. I don't think it's a good idea to do it for JSON. It should at least be configurable option.